api_tests/tests/oauth/test_jwt_unattended_access.py

```diff
@@ -431,3 +468,91 @@ def test_invalid_jwks_resource_url(self):
             },
             expected_status_code=500
         )
+
+    @pytest.mark.apm_1701
+    @pytest.mark.happy_path
+    @pytest.mark.asyncio
+    @pytest.mark.parametrize('product_1_scopes, product_2_scopes', [
+        # Scenario 1: Valid scope set, single product
+        (
+            ['urn:nhsd:apim:app:jwks:personal-demographics'],
+            []
+        ),
+        # Scenario 2: Multiple valid scopes set, single product
+        (
+            ['urn:nhsd:apim:app:jwks:personal-demographics', 'urn:nhsd:apim:app:jwks:ambulance-analytics'],
+            []
+        ),
+        # Scenario 3: Single valid scopes set, different products
+        (
+            ['urn:nhsd:apim:app:jwks:personal-demographics-service'],
+            ['urn:nhsd:apim:usr:aal3:personal-demographics-service']
+        ),
+        # Scenario 4: Multiple valid scopes set, multiple products
+        (
+            ['urn:nhsd:apim:app:jwks:personal-demographics-service'],
+            ['urn:nhsd:apim:app:jwks:ambulance-analytics']
+        )
+    ])
+    async def test_valid_application_restricted_scope_combination(
+        self,
+        product_1_scopes,
+        product_2_scopes,
+        test_app_and_product
+    ):
+        test_product, test_product2, test_app = test_app_and_product
+
+        await test_product.update_scopes(product_1_scopes)
+        await test_product2.update_scopes(product_2_scopes)
+
+        config.JWT_APP_KEY = test_app.get_client_id()
+        jwt = self.oauth.create_jwt(kid='test-1')
+        response = self.oauth.get_jwt_token_response(jwt)
+        assert list(response[0].keys()) == ['access_token', 'expires_in', 'token_type']
+        assert response[1] == 200
+
+    @pytest.mark.apm_1701
+    @pytest.mark.errors
+    @pytest.mark.asyncio
+    @pytest.mark.parametrize('product_1_scopes, product_2_scopes', [
```

Comment:
Might want to consult with @luckmanali, but I think this could use some quite dishonest scenarios, e.g. junk scopes, empty strings, all sorts of horrible things.

Reply:
Some scopes to consider:
["ThisDoesNotExist"] --> a scope that does not exist
["#£$?!&%*.;@~_-"] --> try some special characters to see if it falls over
[""] --> empty string
[None] --> null object
["   urn:nhsd:apim:app:jwks:personal-demographics   "] --> valid but with trailing and leading white spaces